azurelinuxagent/common/osutil/mariner.py

```diff
@@ -0,0 +1,99 @@
+import time  # pylint: disable=W0611
+import base64  # pylint: disable=W0611
+import errno
+import azurelinuxagent.common.conf as conf
+import azurelinuxagent.common.logger as logger  # pylint: disable=W0611
+import azurelinuxagent.common.utils.fileutil as fileutil
+import azurelinuxagent.common.utils.shellutil as shellutil
+import azurelinuxagent.common.utils.textutil as textutil  # pylint: disable=W0611
+from azurelinuxagent.common.osutil.default import DefaultOSUtil
+from azurelinuxagent.common.exception import OSUtilError
+
+class MarinerOSUtil(DefaultOSUtil):
+    def __init__(self):
+        super(MarinerOSUtil, self).__init__()
+        self.agent_conf_file_path = '/etc/waagent.conf'
+        self.jit_enabled = True
+
+    def is_dhcp_enabled(self):
+        return True
+
+    def start_network(self) :
+        return shellutil.run("systemctl start systemd-networkd", chk_err=False)
+
+    def restart_if(self, ifname=None, retries=None, wait=None):
+        shellutil.run("systemctl restart systemd-networkd")
+
+    def restart_ssh_service(self):
+        shellutil.run("systemctl restart sshd")
+
+    def stop_dhcp_service(self):
+        return shellutil.run("systemctl stop systemd-networkd", chk_err=False)
+
+    def start_dhcp_service(self):
+        return shellutil.run("systemctl start systemd-networkd", chk_err=False)
+
+    def start_agent_service(self):
+        return shellutil.run("systemctl start {0}".format(self.service_name), chk_err=False)
+
+    def stop_agent_service(self):
+        return shellutil.run("systemctl stop {0}".format(self.service_name), chk_err=False)
+
+    def register_agent_service(self):
+        return shellutil.run("systemctl enable {0}".format(self.service_name), chk_err=False)
+
+    def unregister_agent_service(self):
+        return shellutil.run("systemctl disable {0}".format(self.service_name), chk_err=False)
+
+    def get_dhcp_pid(self):
+        return self._get_dhcp_pid(["pidof", "systemd-networkd"])
+
+    def conf_sshd(self, disable_password):
+        pass
+
+    def del_root_password(self):
+        try:
+            passwd_file_path = conf.get_passwd_file_path()
+            try:
+                passwd_content = fileutil.read_file(passwd_file_path)
+                if not passwd_content:
+                    # Empty file is no better than no file
+                    raise IOError(errno.ENOENT, "Empty File", passwd_file_path)
+            except (IOError, OSError) as file_read_err:
+                if file_read_err.errno != errno.ENOENT:
+                    raise
+                new_passwd = ["root:*LOCK*:14600::::::"]
+            else:
+                passwd = passwd_content.split('\n')
+                new_passwd = [x for x in passwd if not x.startswith("root:")]
+                new_passwd.insert(0, "root:*LOCK*:14600::::::")
+            fileutil.write_file(passwd_file_path, "\n".join(new_passwd))
+        except IOError as e:  # pylint: disable=C0103
```

Comment:
Please try not to disable pylint errors and rather fix them instead. You can either look at the error code by looking at the ci/3.6.pylintc file or just google the error code, you would get enough examples on how to handle that error. 
(Small note, Pylint errors are mainly either just syntactical errors or beautification errors, but they help maintain the code better so that's why we push for having them fixed rather than disabled :) )

Sample Eg: 
From the ci/3.6.pylintrc file - 
`invalid-name<C0103>: (needs review) Used when the name doesn't conform to naming rules associated to its type (constant, variable, class...).`

And a simple way to fix it is just - 
```python
 except IOError as **error**: 
```

Reply:
updated